### ISSUE TITLE
Create a separate branch for remote server code 

### ISSUE THREAD

**Issue by CandyZQ** [CSbyUs/csbyus-on-rails#7]
Currently, code in master branch is used for both local servers and the remote server. The downside of this is that we need to change Gemfile, potentially leading to different versions of Gemfiles locally. The problem is especially common if some packages are used in one branch but not the other. Different Gemfiles make debug and code-review error-prone. 

It is nice if we could create a branch specifically for remote server and leave the master branch purely for development.

**Comment by tannerljohnson** [comment]
agreed! @CandyZQ do you have any ideas for automating this flow? or streamlining it if not automated?

**Comment by CandyZQ** [comment]
https://www.atlassian.com/blog/git/git-automatic-merges-with-server-side-hooks-for-the-win
This blog here explains how to automate the process locally, but I cannot find any information for auto-merge with Github or Heroku.
